core/src/machine/ppu.rs

```diff
@@ -0,0 +1,251 @@
+use crate::{
+    env::Display,
+    primitives::{Byte, Word, Memory},
+};
+
+
+/// Pixel processing unit.
+pub(crate) struct Ppu {
+    pub vram: Memory,
+    pub oam: Memory,
+
+    /// How many cycles did we already spent in this line?
+    cycle_in_line: u8,
+
+    // ===== Registers ======
+    /// FF40: LCDC
+    lcd_control: Byte,
+
+    /// FF41: LCD status
+    stat: Byte,
+
+    /// FF42: y scroll position of background
+    scroll_y: Byte,
+
+    /// FF43: x scroll position of background
+    scroll_x: Byte,
+
+    /// FF44: LY. Stores the line we are currently drawing (including v-blank
+    /// lines). This value is always between 0 and 154 (exclusive).
+    current_line: Byte,
+
+    /// FF45: LY compare. Is compared to `current_line` all the time. If both
+    /// values are equal, things happen.
+    lyc: Byte,
+
+    /// FF4A: Y window position
+    win_y: Byte,
+
+    /// FF4B: X window position
+    win_x: Byte,
+}
+
+impl Ppu {
+    pub(crate) fn new() -> Self {
+        Self {
+            vram: Memory::zeroed(Word::new(0x2000)),
+            oam: Memory::zeroed(Word::new(0xA0)),
+
+            cycle_in_line: 0,
+
+            lcd_control: Byte::zero(),
+            stat: Byte::zero(),
+            scroll_y: Byte::zero(),
+            scroll_x: Byte::zero(),
+            current_line: Byte::zero(),
+            lyc: Byte::zero(),
+            win_y: Byte::zero(),
+            win_x: Byte::zero(),
+        }
+    }
+
+    /// Loads a byte from VRAM at the given (absolute!) address.
```

Comment:
I think we shouldn't use the absolute address here. The PPU doesn't know about the memory mapping, so the part of subtracting `0x8000` should stay in `mm.rs`. The same holds for `store_vram_byte`, `load_io_byte ` and `store_io_byte `.